src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/EnclosingTypeMapNode.cs

```diff
@@ -0,0 +1,70 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+using Internal.Text;
+using Internal.TypeSystem.Ecma;
+using System.Diagnostics;
+using System.Reflection;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using Internal.ReadyToRunConstants;
+
+namespace ILCompiler.DependencyAnalysis.ReadyToRun
+{
+    public class EnclosingTypeMapNode : ModuleSpecificHeaderTableNode
+    {
+        private MetadataReader _metadata;
+
+        public EnclosingTypeMapNode(EcmaModule module) : base(module)
+        {
+            _metadata = module.MetadataReader;
+            // This map is only valid for assemblies with <= 0xFFFE types defined within
+            if (!IsSupported(_metadata))
+                throw new InternalCompilerErrorException("EnclosingTypeMap made for assembly with more than 0xFFFE types");
+        }
+
+        public static bool IsSupported(MetadataReader metadata)
+        {
+            // This map is only valid for assemblies with <= 0xFFFE types defined within
+            // and really shouldn't be generated for tiny assemblies, as the map provides very little to no value
+            // in those situations
+            int typeDefinitionCount = metadata.TypeDefinitions.Count;
+
+            return ((typeDefinitionCount > 10) && (typeDefinitionCount <= 0xFFFE));
+        }
+
+        public override int ClassCode => 990540812;
```

Comment:
What is this value?

Reply:
Its a random number. It needs to be different from other random numbers used as ClassCode, but that's the only requirement.